{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 61,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "ename": "AttributeError",
     "evalue": "'str' object has no attribute 'append'",
     "output_type": "error",
     "traceback": [
      "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[0;31mAttributeError\u001b[0m                            Traceback (most recent call last)",
      "\u001b[0;32m<ipython-input-61-4e0c338f4efb>\u001b[0m in \u001b[0;36m<module>\u001b[0;34m()\u001b[0m\n\u001b[1;32m     78\u001b[0m                 \u001b[0mgc_count\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0;34m(\u001b[0m\u001b[0mfragment\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mcount\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m'G'\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m+\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mfragment\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mcount\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m'C'\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m     79\u001b[0m                 \u001b[0;31m#gc_count +=1\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m---> 80\u001b[0;31m                 \u001b[0mfragment_gc\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mappend\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mgc_count\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m     81\u001b[0m         \u001b[0mprint\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mfragment_gc\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m     82\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;31mAttributeError\u001b[0m: 'str' object has no attribute 'append'"
     ]
    }
   ],
   "source": [
    "#load the system module\n",
    "\n",
    "#check arguments\n",
    "import sys\n",
    "if len(sys.argv) <3:\n",
    "    print(sys.argv[0] + \": requires .fasta and .gff files\")\n",
    "    sys.exit()\n",
    "\n",
    "#read arguments\n",
    "fsa=sys.argv[1]\n",
    "gff=sys.argv[2]\n",
    "\n",
    "\n",
    "#declare the file names\n",
    "gff_file = 'watermelon.gff'\n",
    "fsa_file = 'watermelon.fsa'\n",
    "\n",
    "#open the files for reading\n",
    "gff_in = open(gff_file, 'r')\n",
    "fsa_in = open(fsa_file, 'r')\n",
    "\n",
    "#declare the variable that will hold the genome sequence\n",
    "genome = ''\n",
    "\n",
    "#initialize a line counter\n",
    "line_number = 0\n",
    "\n",
    "\n",
    "#read in the genome file\n",
    "for line in fsa_in:\n",
    "    #print (str(line_number) + \": \" + line)\n",
    "    \n",
    "    #remove newline\n",
    "    line = line.rstrip ('\\n')\n",
    "    \n",
    "    if line_number > 0:\n",
    "           genome = genome + line\n",
    "        \n",
    "    line_number += 1\n",
    "\n",
    "    #check content after removing first line\n",
    "    #print(genome)\n",
    "    \n",
    "#close the genome file\n",
    "fsa_in.close()\n",
    "\n",
    "#length of genome\n",
    "genome_length=len(genome)\n",
    "#print(genome_length)\n",
    "\n",
    "\n",
    "\n",
    "#read the GFF file\n",
    "for line in gff_in:\n",
    "    #remobe newline\n",
    "    line = line.rstrip('\\n')\n",
    "    \n",
    "    fields = line.split('\\t')\n",
    "    type = fields[2]\n",
    "    start = int(fields[3])\n",
    "    end = int(fields[4])\n",
    "    \n",
    "    #print(type, \"\\t\", start, \"\\t\", end)\n",
    "    \n",
    "    #extract this feature from the genome\n",
    "    fragment = genome[start-1:end]\n",
    "    \n",
    "    #print(fragment)\n",
    "    \n",
    "    if type == 'CDS':\n",
    "        cds += fragment\n",
    "        \n",
    "        gc_count = 0\n",
    "        \n",
    "        for fragment_gc in fragment:\n",
    "            #GC count for each fragment\n",
    "            if gc_count >= 0:\n",
    "                gc_count = (fragment.count('G'))+(fragment.count('C'))\n",
    "                #gc_count +=1   \n",
    "                fragment_gc.append(gc_count)\n",
    "        print(fragment_gc)\n",
    "    \n",
    "    \n",
    "        \n",
    "    \n",
    "#    if type == 'intron':\n",
    "#        intron += fragment\n",
    "    \n",
    "#    if type == 'misc_feature':\n",
    "#        misc += fragment\n",
    "    \n",
    "#    if type == 'repeat_region':\n",
    "#        repeats += fragment\n",
    "        \n",
    "#    if type == 'rRNA':\n",
    "#        rrna += fragment\n",
    "    \n",
    "#    if type == 'tRNA':\n",
    "#        trna += fragment\n",
    "#\n",
    "#print the output\n",
    "#print(cds.count('G'))\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.6.0"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
